### ISSUE TITLE
Regression: Disabling `multi_threaded` feature causes broken rendering

### ISSUE THREAD

**Issue by BigWingBeat** [bevyengine/bevy#14993]
## Bevy version

- Issue occurs on current git main (ffe0f7f)
- Git bisect identifies `d9527c101c2f49c6884763cce36ea1d27dd6a597` (#14833) as the first bad commit
- Issue **does not** occur on crates.io v0.14.1

## \[Optional\] Relevant system information

- `cargo 1.80.1 (376290515 2024-07-16)`
- `Pop!_OS 22.04 LTS`
- `AdapterInfo { name: "AMD Radeon RX 6600 XT (RADV NAVI23)", vendor: 4098, device: 29695, device_type: DiscreteGpu, driver: "radv", driver_info: "Mesa 24.0.3-1pop1~1711635559~22.04~7a9f319", backend: Vulkan }`

## What you did

- Cargo.toml:
```toml
[package]
name = "testy"
version = "0.1.0"
edition = "2021"

[dependencies]
bevy = { git = "https://github.com/bevyengine/bevy", branch = "main", default-features = false, features = [
	"x11",
	"bevy_core_pipeline",
] }
```

- main.rs:
```rust
use bevy::prelude::*;

fn main() -> AppExit {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_systems(Startup, |world: &mut World| {
            world.spawn(Camera3dBundle::default()); // Camera2dBundle also works
        })
        .run()
}
```

## What went wrong

![1](https://github.com/user-attachments/assets/b487fd34-0a17-43b0-998c-7838c1213e6d)

![3](https://github.com/user-attachments/assets/8b5657ad-226a-43e0-8a30-da951ddadcca)

## Additional information

Changing the above Cargo.toml to the following fixes the issue:
```diff
[package]
name = "testy"
version = "0.1.0"
edition = "2021"

[dependencies]
bevy = { git = "https://github.com/bevyengine/bevy", branch = "main", default-features = false, features = [
	"x11",
	"bevy_core_pipeline",
+	"multi_threaded",
] }
```

Additionally, all other features seem to have no effect on it. You can manually re-enable all of the default features except `multi_threaded`, and the issue still occurs. Then enable `multi_threaded`, and it's fixed.

**Comment by alice-i-cecile** [comment]
Thanks for the report! The next step is to bisect the git history to try and figure out which commit is responsible.

**Comment by BigWingBeat** [comment]
Git bisect says `d9527c101c2f49c6884763cce36ea1d27dd6a597 is the first bad commit`, which is #14833.

**Comment by alice-i-cecile** [comment]
Fantastic. FYI @tychedelia <3

**Comment by tychedelia** [comment]
Whoa, huh, will look into this.

**Comment by tychedelia** [comment]
Cannot reproduce on Windows with:

```
AdapterInfo { name: "NVIDIA GeForce RTX 4090", vendor: 4318, device: 9860, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "546.12", backend: Vulkan }
```

But I have an idea and will continue to dig.

**Comment by JMS55** [comment]
Broken rendering like this should never happen. Sounds like wgpu messed up the synchronization somewhere.

**Comment by tychedelia** [comment]
@BigWingBeat Are you seeing this behavior in a more non-minimal repro? One thing I'm confused about is that on Windows and macOS, a window doesn't even open with this configuration. It seems like the x11 feature is causing a window to be created which maybe is an unrelated bug, but just want to confirm that you are seeing issues outside of the repro.

**Comment by BigWingBeat** [comment]
> @BigWingBeat Are you seeing this behavior in a more non-minimal repro? One thing I'm confused about is that on Windows and macOS, a window doesn't even open with this configuration. It seems like the x11 feature is causing a window to be created which maybe is an unrelated bug, but just want to confirm that you are seeing issues outside of the repro.

Yes, broadly speaking it seems like the issue occurs with any bevy project that has an open window, a spawned camera, and `multi_threaded` disabled. This includes most of Bevy's built-in examples, as long as you also re-enable whatever other features the examples need.

You should be able to just enable the `bevy_winit` feature which I assume is what makes a window open. Also note that sometimes you need to re-run the app several times before the visual artifacts start appearing, although sometimes they appear immediately.

**Comment by tychedelia** [comment]
@BigWingBeat will you cherry-pick the fix from https://github.com/bevyengine/bevy/pull/15000 and see if that fixes things for you?

**Comment by BigWingBeat** [comment]
> @BigWingBeat will you cherry-pick the fix from #15000 and see if that fixes things for you?

Yeah, that fixed it!
